lib/request.js

```diff
@@ -2,9 +2,17 @@ const request = require('request');
 const TIMEOUT = 10000;
 
 class Request {
-  constructor(siteid, apikey, defaults) {
-    this.siteid = siteid;
-    this.apikey = apikey;
+  constructor(auth, defaults) {
+    if (typeof auth === 'object') {
```

Comment:
The request class:

- takes an object with siteid and apikey, or a string that's just the app key
- does the exact same thing for both types of authentication
- is only used internally

I'm not a huge fan of switching between app keys and siteid/apikey based on the type of the object being passed in, but the other alternatives weren't great either. If you have any ideas, let me know!

Reply:
Ultimately your solution is fine, but if it's only used internally, we can make larger changes here. For example, we could have a base `Request` class and create `ApiKeyRequest` and `AppKeyRequest` classes, which accept different params, but extend from the base.

Reply:
I don't think either have a common base, since you can't do the same things with the track API that you can with the app API, or vice versa. In the future, however, when the track API accepts api keys, then the old client could extend the base class with the older authentication mechanism